dev/Common/CornerRadiusFilterConverter.cpp

```diff
@@ -51,7 +51,14 @@ winrt::IInspectable CornerRadiusFilterConverter::Convert(
     winrt::hstring const& language)
 {
     auto cornerRadius = unbox_value<winrt::CornerRadius>(value);
-    auto filterType = Filter();
+
+    const auto scale = Scale();
```

Comment:
>const auto scale = Scale(); [](start = 4, length = 27)

should we guard against weird double values? (negative numbers, inf, NaN, and 0)

Reply:
NaN in particular seems like something we might want to check for. what happens if you set a corner radius to NaN? is it like width and height where this is a proxy for Auto?

---
In reply to: [338650803](https://github.com/microsoft/microsoft-ui-xaml/pull/1485#discussion_r338650803) [](ancestors = 338650803)

Reply:
Setting NaN on CornerRadius throws an exception. For this "Scale" property, I think NaN translates to don't apply scale, so I added a guard, if the value is NaN we'll just use the original CornerRadius value.

For negative values, CornerRadius throws exception when it sees negative, I guess I'll just defer the responsibility to CornerRadius. Inf will eventually overflow to negative, so it should be the same story.

Reply:
Could you not default to 0 when it attempts to move the value into a negative?

Reply:
Technically CornerRadius can be negative, it's up to the consumer of CornerRadius to decide how to handle it. We probably shouldn't force it to 0. 

If we do want to default to 0, that should be done in CornerRadius class, the converter should be minimal I think.